leetcode/BinaryTreeTests/BinaryTree_1530.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace BinaryTreeTests;
+
+[TestFixture]
+internal class BinaryTree_1530
+{
+    class Solution {
+        private int _result;
+    
+        public int CountPairs(TreeNode? root, int distance) {
+            _result = 0;
+            PostOrder(root, distance);
+            return _result;
+        }
+
+        private int[] PostOrder(TreeNode? node, int distance) {
+            var distanceCache = new int[distance+1];
+            if(node is null) return distanceCache;
+            // node is leaf node, no child, just return
+            if (node.left == null && node.right == null) {
+                distanceCache[1] = 1; //parent node to this node have distance 1
+                return distanceCache;
+            }
+            var leftDistanceCache = PostOrder(node.left, distance);
+            var rightDistanceCache = PostOrder(node.right, distance);
+
+            for(var l = 1; l<distance; l++) {
+                for(var r=1; r<distance; r++) {
+                    var goodDist = l+r;
+                    if (goodDist <= distance) {
+                        var goodDistCombination = leftDistanceCache[l] * rightDistanceCache[r];
+                        _result += goodDistCombination;
+                    }
```

Comment:
For this node, for this iteration. We want to get
1. left leaves to this node with distance `l`
2. right leaves to this node with distance `r` 
3. distance `l` + `r` not bigger than distance limit
4. get the count of left leaves with distance `l`
5. get the count of right leaves with distance 'r'
6. now 4 and 5 have multiply combination
7. Add that combination to result.